covariate-shift/covariate-shift.whizzml

```diff
@@ -0,0 +1,162 @@
+                            "out_of_bag" oob
+                            "seed" seed}))
+
+;; split-dataset
+;;
+;; splits dataset into two new datasets
+;;
+;; Inputs:
+;;   dst-id: (string) Dataset ID for the dataset we want to split
+;;   rate: (float)  See "sample-dataset"
+;;   seed: (string) See "sample-dataset"
+;;
+;; Output: (list) A list containing two Dataset IDs.
+;;                Corresponds to both pieces of a dataset split.
+
+(define (split-dataset dst-id rate seed)
+  (list (sample-dataset dst-id rate false seed)
+        (sample-dataset dst-id rate true seed)))
+
+
+;; model-evaulation
+;;
+;; creates an evaluation of the given model against the given dataset.
+;;
+;; Inputs:
+;;  model-id: (string) Model ID
+;;  dst-id: (string) Dataset ID
+
+(define (model-evaluation model-id dst-id)
+  (create-and-wait-evaluation {"model" model-id "dataset" dst-id}))
+
+;; avg-phi
+;;
+;; gets the property "average_phi" from the given evaluation
+;;
+;; Input: ev-id (string) Evaluation ID
+;; Output: (float) phi-score for evaluation
+
+(define (avg-phi ev-id)
+  (get-in (fetch ev-id) ["result" "model" "average_phi"]))
+
+;; sample-and-score
+;;
+;; Given a dataset with mixed rows labeled "Training" and "Origin"
+;; makes a test/training split on the dataset
+;; creates a model with the training dataset
+;; evaluates the model witht the test dataset
+;; returns the phi score of the evaulation
+;;
+;; Inputs: combined-dst: (string) Dataset ID
+;;         seed: (string) Determinisitic seed
+;;
+;; Output: (float) phi-score for evaluation
+;;
+
+(define (sample-and-score combined-dst seed)
+  (let (ids (split-dataset combined-dst 0.8 seed)
+        model (create-and-wait-model {"dataset" (nth ids 0)
+                                      "objective_field" "Origin"})
+        ev-id (model-evaluation model (nth ids 1)))
+      (avg-phi ev-id)))
+
+;; multi-phis
+;;
+;; Creates a dataset with mixed rows labeled "Training" and "Origin"
+;; runs the `sample-and-score` function `n` times, using a generated seed.
+;;
+;; Inputs:
+;;   n: (integer) Number of times to run the phi-measure function
+;;   train-dst: (string) Dataset ID for the training data
+;;   prod-dst: (string) Dataset ID for the production data
+;;
+;; Output: (map) Shows average and list of results for the phi-measure function
+;;    e.g., {"average": 0.032 "list": [ -0.1293 0.0983 ... ]}
+
+(define (multi-phis n train-dst prod-dst)
+  (let (comb-data (combined-data train-dst prod-dst))
+    (loop (seeds (range 0 n) out [])
+      (if (= [] seeds)
+        {"list" out
+         "average" (/ (reduce + 0 out) (count out))}
+        (recur (tail seeds)
+               (append out (sample-and-score
+                             comb-data
+                             (str "test-" (head seeds)))))))))
+
+(define output (multi-phis n train-dst prod-dst))
+
+;; EXAMPLE
+;;
+;; (multi-phis 3
+;;    "dataset/56c3af3f7e0a8d6cca01292e""dataset/56c3af3f7e0a8d6cca01292e")
```

Comment:
this example suggests that using the same dataset for train and prod is a good idea... is it?